### DISCUSSION TITLE
Bugs

### DISCUSSION THREAD

**Discussion by ph1p** [ph1p/figma-measure#16] [General]
If you found any bugs, just post them here.

**Comment by glowdanny** [comment]
Hi there - for some reason when I try to add a measurement, nothing happens. I have measurements that work on another page. Does the plugin only support 1 page currently?

**Reply by ph1p** [reply]
Hello! I am currently working on a solution for the groups that meets all requirements. The plugin needs the group to be performant, as it has to iterate over many levels if it wasn't there. Hopefully I can provide it in the next few days (:

**Reply by ph1p** [reply]
@glowdanny now it should be fixed :)

**Reply by glowdanny** [reply]
Awesome - thank you!

**Reply by ph1p** [reply]
You're welcome ;)

**Comment by markmanfrey** [comment]
I'm getting inverted text on some of my measurements. Any tips?

**Reply by ph1p** [reply]
Hi @markmanfrey , sounds like your element is rotated.  It would be cool if you could invite me to a file or create a file where you reproduced it. So I can take a look :)

**Reply by markmanfrey** [reply]
done. The example is waaaay off to the right

**Reply by markmanfrey** [reply]
Isolated the issue into a new figma file for you. Updated your invite

**Reply by markmanfrey** [reply]
I don't believe it's a rotation. If I flip the text both vertically and horizontally, then it reads correctly. Seems like an inverted scaling or something.

**Reply by ph1p** [reply]
So I played around with your component and it took me a while to figure out what was wrong :D The bottom part of your input field is mirrored vertically. I think that's because it's a copy of the top half of the input field. Unfortunately, I can't tell in Figma if an element is mirrored or not :/ The easiest solution would be to copy the top half again and only rotate or mirror the children.

Through you I found a few minor bugs though, thanks :) And maybe in the future I can use the Figma API to see if something is mirrored.

**Reply by markmanfrey** [reply]
Which part of the lower section is mirrored? I can't see it described in the side panel. I'm sure you're right, but how did you figure that out? Query it via code?

**Reply by ph1p** [reply]
That's my problem too, you can't see which element is mirrored/flipped in Figma. The shortcuts are `shift+v` and `shift+h`.

![screen](https://user-images.githubusercontent.com/15351728/150180212-4cfc9e14-9f25-4186-84dc-443802ba70f3.png)

**Reply by markmanfrey** [reply]
Is the rotation indicative of a mirror? Also, is there a way to orient the text independent of the object?

**Comment by rdsalves** [comment]
Hey @ph1p , first, amazing work, I really appreciate.

I saw today the last update and there is something strange happening, the measures are blending inside the components.
Take a look at the picture

I tried to frame the icon, and the measure dot line got inside the component.
![Screen Shot 2022-01-24 at 11 46 48](https://user-images.githubusercontent.com/52305033/150804738-a2551a17-ddb9-49fe-adfe-735c96ddfa99.png)

**Reply by ph1p** [reply]
Hi @rdsalves! Thanks for reporting :) Is it possible to invite me to this file or a duplicated one? It would also be great if you could create a version of what you think it should look like. ([removed mail])

**Reply by ph1p** [reply]
Ok, I found the problem. The frame with auto layout has included the measurement group. Now the measurement group is pushed one level higher. 

This should be fixed now! :)

**Reply by rdsalves** [reply]
Ooohh now I see, understood what you did.
You tried to put the measure close to the object being measured, and that makes the measure got inside the component.
Before the update, all measurements were created in a group, outside the frames.
Now, you are placing this group, inside the respective frame.

**Thanks for solving this so fast.**
Let me show you another thing.

I imagine you made this change because a lot of people complained about the difficulty of moving the measurements along with the material. (which honestly, I thought was silly, if you want to move things, just unlock, but they are too lazy and need to complain about everything)

I even think it makes sense for the group to stay inside the corresponding frame, but this caused another small problem, I'll show you:
![Screen Shot 2022-01-24 at 12 25 32](https://user-images.githubusercontent.com/52305033/150811912-65ed928e-be5f-4efd-88a3-c3efb7028ec5.png)

Can you see? As the measurement is inside the frame, it ends up leaking and is hidden by the sides.

@ph1p  honestly, I don't care, I'm just showing you so you are aware. I can solve this by easily dragging the measurement group out of the frame and overlaying the frame. But soon some lazybones will come to complain about it.

**Reply by ph1p** [reply]
You're welcome :) Yes, I am aware of that, it is due to "content clipping". I could also filter out those frames, but then the group keeps moving up the layer tree and maybe end up in the root. I added this feature because the "global group" was not considered good by many (including me). But it was necessary, because I didn't find any other way to create measurements in a performant way. With large files I have to iterate over many layers and that can't be avoided.

Now all measurements are always as close as possible to their elements, so that they are easier to export. To just show others the hidden measurements you can temporarily turn off the content clipping on the frame or do you think it would be better to also filter these type of parent elements?

Thank you for reporting it though! :) I'll write a small paragraph in the FAQ. (https://github.com/ph1p/figma-measure/blob/master/docs/faq.md)

**Reply by rdsalves** [reply]
Awesome @ph1p you are 100% right, the easiest solution it is to disable Clip Content.
I totally agree with you, the global group was not the best solution, and you should not try to filter out those frames.

I don't have the knowledge to know what this implies from a technical point of view, but filtering always seems to me to be an action that requires processing, and Figma is not the most optimized tool in the world, as it is based on Chromium.

So I suggest keeping it as is, and yes, add the information about Clip Content to the Faq, it's a completely honest solution.

**Reply by ph1p** [reply]
Alrighty, good to hear! :) I've added a small paragraph

**Comment by megaroeny** [comment]
I found a bug today. I have two objects within a small Auto Layout frame. They're spaced `4px` apart, but for some reason the plugin displayed as `3px`. Maybe because the space is so small?

![Screen Shot 2022-04-22 at 3 45 38 PM](https://user-images.githubusercontent.com/1559447/164783742-887d2733-3d9c-4b35-8003-6e83c1c1a414.png)

This is my layer structure:

![image](https://user-images.githubusercontent.com/1559447/164783968-f2077541-85b3-46a0-8b44-b1a6ea3265bc.png)

**Reply by ph1p** [reply]
Hi @megaroeny - Thanks for reporting! Could you please in invite me to your file? (me@ph1p.dev) I'll take a look at it as soon as I can (:

**Reply by megaroeny** [reply]
> Could you please in invite me to your file?

I'll send you an invite next week, or later tonight. I need to make a duplicate first and delete everything else so I don't share our company property/work 🙃

**Reply by ph1p** [reply]
Alrighty! No stress :)

>  I don't share our company property/work

Totally understandable :D 

I'll also try to recreate it by myself

**Reply by ph1p** [reply]
Fixed! This error occurred because of the underlining. I calculate the spacing with a bounding box and there figma includes also the underline. In your design this overhang is exactly 1px. Now I use the bounding box only when the element is rotated :)

**Reply by megaroeny** [reply]
Fantastic! I had a feeling it might have been the line.

**Comment by megaroeny** [comment]
@ph1p I just invited you to an isolated file in my drafts and was able to reproduce the bug. 🙂 No rush fixing this this weekend.

**Comment by Xamantu** [comment]
I found out that the measurement works incorrectly because it considers shadows to be part of the selected elements.

![image](https://user-images.githubusercontent.com/9420003/170061122-023465d0-fbf1-4b8c-be68-153bbb18a669.png)

![image](https://user-images.githubusercontent.com/9420003/170061665-4a94fffe-c462-40df-8187-4c9623b79d35.png)

If I remove the shadows, measurement works perfectly.
![image](https://user-images.githubusercontent.com/9420003/170060795-752513b4-af2a-4abf-8ce3-4240966c1623.png)

**Reply by ph1p** [reply]
Could you please create an open file with a minimal reproduction? I will try to fix it as soon as possible. I have already worked on the shadows a few days ago, but apparently there are still a few constellations that I have not taken into account. Thank you (:

**Reply by Xamantu** [reply]
Sure thing, just invited you!

**Comment by kemie** [comment]
Units decimals are rounded.
I thought I'd be clever and have measurements in rems instead of pixels by using the formula ($/16)rem, but anything smaller than 1 rem is just displayed as 0rem and no decimals are ever shown, which results in incorrect measurements all over. Maybe implement a couple of decimals?

**Reply by ph1p** [reply]
Hi @kemie 
You can write it simply like this: `($##/16)rem`. The number of hash marks determines the number of decimal places, but if it is zero it will not be displayed. 

1.50 -> $## -> 1.5
1.51 -> $## -> 1.51

**Reply by kemie** [reply]
Thanks!!

**Comment by megaroeny** [comment]
I think this might have to do with the latest version of Auto Layout, but I've been encountering numerous issues where the generated measurement is not correct, or is saying "measurement is zero". Here is an example of this happening. I ungrouped the auto layout frame it is in, so it's within the root level of the component (and should be able to calculate 8px padding top/right, but the same result happened.)

[Measure - absolute position.webm](https://user-images.githubusercontent.com/1559447/177592737-4ad088b7-c0ec-47b9-b68c-3e5979aa6c7a.webm)

**Reply by ph1p** [reply]
Thanks for reporting! I'll take a look and fix it as soon as I can. I was the last weeks less on the plugin, because I have a new job and made a little vacation :)
Videos like this help me a lot!

**Reply by megaroeny** [reply]
My solution for now, was to put this component instance (I detached it in this example), in another frame, and draw a square above this dismiss `X` icon button. I was able to "fake" it, and this works for now. 🙂

**Reply by megaroeny** [reply]
> ... because I have a new job and made a little vacation :) Videos like this help me a lot!

Nice! No worries at all, and congrats.

**Reply by megaroeny** [reply]
Here's one more example, of where I was expecting 16px padding, but it's showing 8px for some reason. Maybe has something to do with the position of the dismiss icon button? Not sure. Hope this helps

![Screen Shot 2022-07-06 at 12 10 52 PM](https://user-images.githubusercontent.com/1559447/177595942-4c95cdf5-eb0f-43dc-a378-56f95e443083.png)

**Reply by megaroeny** [reply]
Okay, last update, promise haha! (I keep discovering related things...) I think because I have "Clip content" off on the root level of this component, the "📐 Measurements" group is being included as part of the measurements.

**Reply by ph1p** [reply]
Is it possible for you to copy the file and invite me? Or to copy specific parts with errors? Thanks! me@ph1p.dev

**Reply by megaroeny** [reply]
> Is it possible for you to copy the file and invite me? Or to copy specific parts with errors? Thanks! [me@ph1p.dev](mailto:me@ph1p.dev)

Sure, let me isolate and duplicate the object in a drafts file first. I'll get back to you later today with that.

**Reply by megaroeny** [reply]
@ph1p invite sent! 

_(Deleted my last message, sorry I was doing too many things at the same time and totally missed your email address earlier 🤦🏼 )_

**Reply by megaroeny** [reply]
@ph1p was following-up to see if you ever used my file invite?

**Comment by megaroeny** [comment]
Came across another subtle bug. Think it has to do with being nested, and maybe also to do with the new "single-sided" strokes property. 

You can in the tooltip below that I have set it to `6px` Left side, but it shows as `4`.

![image](https://user-images.githubusercontent.com/1559447/181789171-0af636f9-2a55-4a3d-97e9-9d560bcbfa36.png)

**Reply by ph1p** [reply]
Hi @megaroeny! Sorry for the delay. I unfortunately got corona, so everything else had to wait (:
The error occurs because I always look for the parent element. I have now adjusted it so that if the parent element has an autolayout, it is ignored and continues to go up the tree.
I'm going to test it a little bit now.

You can also do it, if you want :) I built a version for you, you can open it with figma->plugins->development->import from manifest....
[Measure.zip](https://github.com/ph1p/figma-measure/files/9221996/Measure.zip)

**Reply by megaroeny** [reply]
Wow so sorry to hear that. There’s no rush, but sure I can see if I can help with a PR. I haven’t worked on a plug-in in a while but I’m familiar with it since I made one myself. 😅 Thank you!

**Reply by ph1p** [reply]
All is well :) the worst is over. You don't have to do a PR, I think I already fixed the bug. I just wanted to give you the plugin zip so you can click around a bit yourself.

**Reply by ph1p** [reply]
<img width="253" alt="Screenshot 2022-07-30 at 14 39 56" src="https://user-images.githubusercontent.com/15351728/181914687-e7db81c6-29b1-4d6d-86cb-a79506d1d627.png">

In the upcoming version the plugin now also recognizes the stroke directions :)

**Reply by megaroeny** [reply]
Fantastic, thank you so much. Glad I could help too, and please if you ever need someone to test let me know. 😄

**Reply by ph1p** [reply]
Haha :D if you want, here is another build.
I also implemented dark mode, fixed some bugs and added the possibility to save all measurements in a global group again.

[Measure.zip](https://github.com/ph1p/figma-measure/files/9226675/Measure.zip)
